src/data-workspace/data-entry-cell/inner-wrapper.js

```diff
@@ -102,22 +116,37 @@ export function InnerWrapper({
         ? styles.synced
         : null
 
+    // clear error if reset to initital-value
+    useEffect(() => {
+        if (!dirty || synced) {
+            clearSyncError(dataValueParams)
+        }
+    }, [clearSyncError, dirty, dataValueParams, synced])
```

Comment:
Yup, a great point and a bit unfortunate. Our `use<>Params` hooks are not stable, but probably should be. Unfortunately this needs quite a few `useMemo`s to fix properly I think. In `useDataValueParams` and `useApiAttributeParams`. Not sure if I want to tackle that now, since those are used in a lot of places across the app, and I would categorize that has high risk to change.

Reply:
I did make a separate PR to fix this here: https://github.com/dhis2/aggregate-data-entry-app/pull/229